basic_hashtable/b_hashtables.c

```diff
@@ -114,7 +138,12 @@ char *hash_table_retrieve(BasicHashTable *ht, char *key)
  ****/
 void destroy_hash_table(BasicHashTable *ht)
 {
+  for(int i = 0; i < ht->capacity; i++){
```

Comment:
C makes me paranoid :)

something like this is probably best
```
for (int i = 0; i < ht->capacity; i++) {
    if (ht->storage[i] != NULL) {
      destroy_pair(ht->storage[i]);
    }
  }
```

Reply:
The way destroy pair works is that it does nothing for NULL so that's why I didn't include that logic.